### ISSUE TITLE
Move longhaul to the same Azure subscription we use for E2E tests

### ISSUE THREAD

**Issue by artursouza** [dapr/test-infra#167]
Longhaul tests are still running in a subscription that is only accessible by Msft employees. Both, release and nightly environments, should be moved to the same Azure subscription we use for E2E tests.

Child of: https://github.com/dapr/test-infra/issues/156

**Comment by tmacam** [comment]
Listing steps performed as a log of what was done and to serve as reference/documentation in case this needs to be redone.

# Intro

* Applications will live in the `aks-longhaul-release` and `aks-longhaul-weekly` subscription
* We are not using the Azure-hosted grafana or prometheus yet - that's planned for the future.

# Setup environment and credentials

```bash

# From https://github.com/dapr/test-infra/pull/203 and https://github.com/dapr/test-infra/blob/master/README.md

export SUBSCRIPTION_TO_BE_USED=INSERT_SUBSCRIPTION_UUID_HERE
export release_or_weekly='release' # use 'weekly' for weekly
export resourceGroup="aks-longhaul-${release_or_weekly}"
export DAPR_VERSION_TO_INSTALL='1.12.0'
export location=eastus
export clusterName=$resourceGroup
export MONITORING_NS=dapr-monitoring
```

# Login to OSS subs

```bash
# First, loging on Dapr OSS subscription on your default browser

# Then, login on az CLI
az account clear && az login --output=none && az account set --subscription ${SUBSCRIPTION_TO_BE_USED}
```

# Create new subscriptions

```bash
az group create --name ${resourceGroup} --location ${location}
```

# Deploy clusters

```bash
az deployment group create \
    --resource-group ${resourceGroup} \
    --template-file ./deploy/aks/main.bicep \
    --parameters deploy/aks/parameters-longhaul-${release_or_weekly}.json
```

# Remove Dapr AKS extension

```bash
# We want to manually control Dapr setup, so let's remove the Azure-controlled Dapr ext.
az k8s-extension delete --yes \
    --resource-group ${resourceGroup} \
    --cluster-name ${clusterName} \
    --cluster-type managedClusters \
    --name ${clusterName}-dapr-ext
```

# Get cluster credentials

```bash
az aks get-credentials --admin --name ${clusterName} --resource-group ${resourceGroup}
```

# Install latest stable on both clusters through helm

```bash
# Just for good measure...
dapr uninstall -k

# Now to the helm chart upgrade
helm repo update && \
helm upgrade --install dapr dapr/dapr \
    --version=${DAPR_VERSION_TO_INSTALL} \
    --namespace dapr-system \
    --create-namespace \
    --wait
```

# Bounce the apps (we just re-installed Dapr)

```bash
for app in "feed-generator-app" "hashtag-actor-app" "hashtag-counter-app" "message-analyzer-app" "pubsub-workflow-app" "snapshot-app" "validation-worker-app" "workflow-gen-app"; do
    kubectl rollout restart deploy/${app} -n longhaul-test || break
done 
```

# Setup monitoring namespace (next steps require this)

```bash
# From https://github.com/dapr/test-infra/blob/master/.github/workflows/dapr-longhaul-weekly.yml
kubectl get namespace | grep ${MONITORING_NS} || kubectl create namespace ${MONITORING_NS}
```

# Install Prometheus through helm chart

```bash
# Following https://docs.dapr.io/operations/observability/metrics/prometheus/#setup-prometheus-on-kubernetes

helm repo add prometheus-community https://prometheus-community.github.io/helm-charts && \
helm repo update && \
helm install dapr-prom prometheus-community/prometheus \
    --namespace dapr-monitoring \
    --create-namespace \
    --wait
```

# Install Prometheus custom setting

This is being bypassed as we fixed dashboard code in dapr/dapr#7121. There is no need
to install custom prometheus setting. Rejoice.

# Install Grafana through helm chart

```bash
    # https://docs.dapr.io/operations/observability/metrics/grafana/#setup-on-kubernetes
    helm repo add grafana https://grafana.github.io/helm-charts && \
    helm repo update && \
    helm upgrade --install grafana grafana/grafana \
        --values ./grafana-config/values.yaml \
        --namespace ${MONITORING_NS} \
        --create-namespace \
        --wait && \
    kubectl get pods -n ${MONITORING_NS}
```

# Configure grafana

Steps here are basically just following the steps described on  https://docs.dapr.io/operations/observability/metrics/grafana/#configure-prometheus-as-data-source

## Log in to grafana

```bash
kubectl get secret --namespace dapr-monitoring grafana -o jsonpath={.data.admin-password} | base64 --decode | clip.exe
kubectl port-forward svc/grafana 8080:80 --namespace dapr-monitoring
```

## Register prometheus datasource

Just follow https://docs.dapr.io/operations/observability/metrics/grafana/#configure-prometheus-as-data-source

## Import dashboards (from #7121)

Use the code from dapr/dapr#7121 or, if it is merged, from https://github.com/dapr/dapr/blob/master/grafana/

Remember: `cat ... | clip.exe` or ` cat ... | pbcopy` is your friend.


# Pending

- [ ] Create credentials for both clusters
- [ ] Update weekly workflow to work on new cluster
- [ ] Update release workflow to work on new cluster
- [ ] Update secrets on GH with new credentials
- [ ] Verify clusters and workflows are working as expected
- [ ] Remove cluster in internal subscription

**Comment by tmacam** [comment]
Added screenshot of the current status of the dashboard to #210.

I am closing this issue as the transition was done, the only thing pending on this transition is the removal of the old longhaul clusters, which is being tracked on #210 .
